tinygrad/codegen/kernel.py

```diff
@@ -352,7 +352,8 @@ def buf_index(src: LazyOp) -> Optional[int]:
           # TODO: apply tc even if the sources are not from LOAD
           if src.op is BufferOps.LOAD and src.arg.dtype == tc.dtype_in: return self.bufs.index(cast(MemBuffer, src.arg))
           try:
-            if opt_level >= 1 and src.op is UnaryOps.CAST and src.arg[0] == tc.dtype_in: return self.bufs.index(cast(MemBuffer, src.src[0].arg))
+            if opt_level >= 1 and src.op is UnaryOps.CAST and src.arg == tc.dtype_in:
+              return self.bufs.index(cast(MemBuffer, src.src[0].arg))
```

Comment:
nit: why is this two lines?

Reply:
Line was too long with the bitcast, will update it in a sec.